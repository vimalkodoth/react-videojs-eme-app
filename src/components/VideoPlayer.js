import React from "react";
import { css } from "@emotion/core";
import player from "./../utils/player";

class VideoPlayer extends React.Component {
    player = null;

    componentDidMount() {
        this.player = player.init(
            this.videoNode,
            {
                html5: {
                    nativeCaptions: false
                },
                ...this.props
            },
            () => {
                this.player.play();
            }
        );
    }

    componentWillUnmount() {
        if (this.player) {
            this.player.destroy();
            this.player = null;
        }
    }

    render() {
        return (
            <div>
                <div data-vjs-player>
                    <video
                        ref={(node) => (this.videoNode = node)}
                        className="video-js vjs-theme-rakuten full"
                        css={Player}
                    ></video>
                </div>
            </div>
        );
    }
}

export default VideoPlayer;

const Player = css`
    &.vjs-theme-rakuten {
        max-width: 100%;
        &.full {
            position: absolute;
            left: 0;
            right: 0;
            min-width: 100%;
            height: 100%;
        }
        max-width: 100%;
        min-width: 100%;
        .vjs-big-play-button {
            position: absolute;
            top: 50%;
            left: 50%;
            margin-top: -0.81666em;
            margin-left: -1.5em;
        }
    }
`;
