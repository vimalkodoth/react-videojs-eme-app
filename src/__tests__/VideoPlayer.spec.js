import React from "react";
import { render, shallow, mount } from "enzyme";
import VideoPlayer from "./../components/VideoPlayer";

const videoJsOptions = {
    src:
        "https://bitmovin-a.akamaihd.net/content/art-of-motion_drm/mpds/11331.mpd",
    autoplay: true,
    controls: true,
    type: "application/dash+xml",
    keySystems: {
        "com.widevine.alpha": "https://widevine-proxy.appspot.com/proxy"
    }
};

describe("VideoPlayer", () => {
    xit("renders correctly", () => {
        const component = render(<VideoPlayer {...videoJsOptions} />);
        expect(component).toMatchSnapshot();
    });
    it("renders a video element", () => {
        const component = render(<VideoPlayer {...videoJsOptions} />);
        expect(component.find("video").length).toEqual(1);
    });
    it("renders video element in fullscreen", () => {
        const component = render(<VideoPlayer {...videoJsOptions} />);
        expect(component.find(".video-js.full").length).toEqual(1);
    });
});
