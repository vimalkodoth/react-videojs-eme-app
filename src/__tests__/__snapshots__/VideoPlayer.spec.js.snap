// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "data-vjs-player": "true",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "video-js vjs-theme-rakuten full css-jre9ez-Player",
            },
            "children": Array [],
            "name": "video",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-vjs-player": undefined,
        },
        "x-attribsPrefix": Object {
          "data-vjs-player": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
